framework/src/application/application.ts

```diff
@@ -82,29 +83,85 @@ const registerProcessHooks = app => {
 			},
 			'System error: unhandledRejection',
 		);
-		app.shutdown(1, err.message);
+		// eslint-disable-next-line @typescript-eslint/no-floating-promises
+		app.shutdown(1, (err as Error).message);
 	});
 
-	process.once('SIGTERM', () => app.shutdown(1));
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises
+	process.once('SIGTERM', async () => app.shutdown(1));
 
-	process.once('SIGINT', () => app.shutdown(1));
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises
+	process.once('SIGINT', async () => app.shutdown(1));
 
-	process.once('exit', (error, code) => app.shutdown(code, error));
+	// eslint-disable-next-line
+	process.once('exit' as any, (code: number) => app.shutdown(code));
 };
 
-class Application {
-	constructor(genesisBlock, config = {}) {
+interface ApplicationConfig {
+	label: string;
+	version: string;
+	protocolVersion: string;
+	networkId: string;
+	lastCommitId: string;
+	buildVersion: string;
+	ipc: {
+		enabled: boolean;
+	};
+	tempPath: string;
+	readonly forging: {
+		readonly waitThreshold: number;
+		readonly delegates: DelegateConfig[];
+		readonly force?: boolean;
+		readonly defaultPassword?: string;
+	};
+	readonly rebuildUpToRound: string;
+	readonly network: NetworkConfig;
+	constants: {
+		[key: string]: {} | string | number | undefined;
+	};
+	genesisConfig: {};
+	components: {
+		[key: string]: {} | undefined;
+		logger: {
+			logFileName: string;
+		};
+	};
+	modules: ModulesOptions;
+}
+
+export class Application {
+	public logger: Logger;
+	public config: ApplicationConfig;
+	public constants: NodeConstants;
+
+	private _node!: Node;
+	private _network!: Network;
+	private _controller!: Controller;
+	private _applicationState!: ApplicationState;
+	private _transactions: { [key: number]: typeof BaseTransaction };
+	private _modules: { [key: string]: InstantiableModule<BaseModule> };
+	private _channel!: InMemoryChannel;
+
+	private readonly _genesisBlock: GenesisBlockInstance;
+	private _migrations: { [key: string]: object };
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	private readonly storage: any;
```

Comment:
Storage has a particular interface already defined in types, why not to use it?

Reply:
The type was not enough, and needed to add more things to be compatible with all the libraries, but since we are gonna remove it, i thought `any` for now is fine